src/utils.js

```diff
@@ -22,6 +22,19 @@ export function splitLocation(location) {
   };
 }
 
+export function checkStatus(sections) {
+  // If even one open section available, return open
+  for (var i = 0; i < sections.length; i++) {
+    if (sections[i].status === "Open") {
+      return "Open";
+    }
+  }
+  if (sections[0].status === "WaitList") {
+    return "Waitlist";
+  }
+  return sections[0].status;
```

Comment:
I can see there being issues where the first section reads as closed but others may be waitlisted so the whole course appears closed. Also we can use the `array.some` method to find whether some element exists rather than looping through.
```suggestion
  const hasOpenSections = sections.some(({status}) => status === "Open");
  const hasWaitlistSections = sections.some(({status}) => status === "WaitList");
  if (hasOpenSections) {
    return "Open";
  }
  return hasWaitlistSections ? "Waitlist" : "Closed";
```
I personally feel that this reads a bit clearer than what was there before and would account for the potential issue above. 

Also as a disclaimer, this code should be tested and played around with before committing to your own PR.

Reply:
Committed your suggestion!